ui/k-toolbar/main.kytos

```diff
@@ -0,0 +1,284 @@
+          });
+
+      request.done(function(data) {
+        for (element in data.result){
+          let current_switch = self.switches[data.result[element]["dpid"]]
+          let current_interface = current_switch.interfaces[`${data.result[element]["dpid"]}:${data.result[element]["port"]}`]
+          if (!data.result[element].hasOwnProperty('out')){
+            data.result[element].out = {"port": "NA", "vlan": "NA"};
+          } else if (current_interface.name != undefined && current_interface.name != ""){
+            let current_out_interface = current_switch.interfaces[`${data.result[element]["dpid"]}:${data.result[element]["out"]["port"]}`]
+            data.result[element]["out"]["port"] = current_out_interface["name"]
+          }
+          if (current_switch.metadata.node_name != undefined && current_switch.metadata.node_name != ""){
+            data.result[element]["dpid"] = current_switch.metadata.node_name
+          }
+          if (current_interface.name != undefined && current_interface.name != ""){
+            data.result[element]["port"] = current_interface.name
+          }
+        }
+        self.trace = data;
+        self.show();
+      });
+
+      request.fail(function(data) {
+        let notification = {
+          icon: 'gear',
+          title: 'Bad request',
+          description: data.status + ': ' + data.responseJSON.description
+        };
+        self.$kytos.$emit("setNotification", notification);
+      });
+    },
+    show() {
+      var content = {
+                    "component": 'amlight-sdntrace_cp-k-info-panel-show_trace_results',
+                    "content": this.trace,
+                    "icon": "search-location",
+                    "title": "Trace",
+                    "subtitle": "by amlight/sdntrace_cp"
+                    }
+      this.$kytos.$emit("showInfoPanel", content)
+      },
+    get_topology(){
+      var self = this
+      $.ajax({
+        async: true,
+        dataType: "json",
+        url: this.$kytos_server_api + "kytos/topology/v3",
+
+        success: function(data) {
+          self.switches = data['topology']['switches']
+          self.links = data['topology']['links']
+        }
+      });
+    },
+    reset_inputs() {
+    this.dpid = ""
+    this.in_port = ""
+    this.eth.dl_vlan = ""
+    this.eth.dl_type = ""
+    this.eth.dl_src = ""
+    this.eth.dl_dst = "" 
+    this.ip.nw_src = "" 
+    this.ip.nw_dst = ""
+    this.ip.ipv6_src = ""
+    this.ip.ipv6_dst = ""
+    this.ip.nw_proto = ""
+    this.ip.nw_tos = ""
+    this.tp.tp_src = ""
+    this.tp.tp_dst = ""
+    }
+    },
+    computed: {
+      get_dpids(){
+      var dpids = []
+      $.each(this.switches, function(key, value){
+          description = value.dpid
+          if (value.metadata.nodename != undefined && value.metadata.nodename != "") {
+            dpids.push(`${value.metadata.node_name} - ${value.dpid}`)
+          } else {
+            dpids.push(value.dpid)
```

Comment:
Here it'd be cool to have `${value.data_path - value.dpid}`, typically the `data_path` value is meaningful and helpful to identify the switch when `node_name` isn't define. Check out [this comment here](https://github.com/kytos-ng/sdntrace_cp/pull/111#discussion_r1481520357)